Podcast/SettingsPageViewController.swift

```diff
@@ -0,0 +1,207 @@
+//
+//  SettingsPageViewController.swift
+//  Podcast
+//
+//  Created by Drew Dunne on 10/22/17.
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+
+// Add more field types as created
+// String value is cell reuse ID
+enum SettingsFieldType {
+    case button
+    case textField
+    case disclosure
+    case label
+    case toggle
+    
+    static let allValues: [SettingsFieldType] = [.button, .textField, .disclosure, .label, .toggle]
+    
+    var reuseIdAndClass: (String, AnyClass) {
+        switch self {
+        case .textField:
+            return ("textFieldSettingsCell", TextFieldSettingsTableViewCell.self)
+        default:
+            return ("settingsCell", SettingsTableViewCell.self)
+        }
+    }
+}
+
+/**
+ * Create a section in settings by setting these
+ */
+struct SettingsSection {
+    var header: String?
+    var footer: String?
+    var items: [SettingsField]
+    
+    init(items: [SettingsField]) {
+        self.items = items
+    }
+}
+
+/**
+ * Each on of these in the items array of a section will be shown in that section
+ */
+struct SettingsField {
+    var title: String
+    var saveFunction: ((Any) -> Void)
+    var type: SettingsFieldType
+    var placeholder: String // Only used for TextField!
+    var tapAction: (() -> Void) // User for disclosure, but applies to all
+    
+    init(title: String, saveFunction: @escaping ((Any) -> Void) = {_ in }, type: SettingsFieldType = .label, placeholder: String = "", tapAction: @escaping (() -> Void) = {}) {
+        self.title = title
+        self.saveFunction = saveFunction
+        self.type = type
+        self.placeholder = placeholder
+        self.tapAction = tapAction
+    }
+}
+
+/**
+ * SettingsPageViewController is a settings controller where changing the sections and items
+ * array will dynamically create a settings view with desired actions.
+ */
+class SettingsPageViewController: ViewController, UITableViewDelegate, UITableViewDataSource {
+
+    // View items
+    private var tableView: UITableView!
+    private var saveButton: UIBarButtonItem!
+    
+    // Can change this but not recommended
+    var sectionSpacing: CGFloat = 18
+    
+    // Settable configurations
+    var sections: [SettingsSection] = []
+    var showSave: Bool! {
+        didSet {
+            if showSave {
+                navigationItem.rightBarButtonItem = saveButton
+            } else {
+                navigationItem.rightBarButtonItem = nil
+            }
+        }
+    }
+    var returnOnSave:Bool = true
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationController?.setNavigationBarHidden(false, animated: false)
```

Comment:
is this not the default? or do we need to do this because we are coming from internal profile?

Reply:
Yeah it's necessary because of some weird stuff in profile. I've been looking into a way to fix this.